sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileWrite.scala

```diff
@@ -91,9 +92,10 @@ trait FileWrite extends Write {
       throw new IllegalArgumentException("Expected exactly one path to be specified, but " +
         s"got: ${paths.mkString(", ")}")
     }
-    val pathName = paths.head
-    SchemaUtils.checkColumnNameDuplication(
-      schema.fields.map(_.name).toImmutableArraySeq, caseSensitiveAnalysis)
+    if (!allowDuplicatedColumnNames) {
+      SchemaUtils.checkColumnNameDuplication(
+        schema.fields.map(_.name).toImmutableArraySeq, caseSensitiveAnalysis)
```

Comment:
Shall we leave a warning message?

Reply:
if a format allows duplicated names, then we don't need to warn users as it's a valid operation.